packages/map-template/src/components/LocationDetails/LocationDetails.jsx

```diff
@@ -52,8 +73,46 @@ function LocationDetails({ onBack, onStartWayfinding, onSetSize, snapPointSwiped
         if (location) {
             locationInfoElement.current.location = location;
             setLocationDisplayRule(mapsIndoorsInstance.getDisplayRule(location));
+            setDestinationLocation(location)
+
+        }
+
+        if (currentKioskLocation) {
+            setOriginLocation(currentKioskLocation)
+        }
+    }, [location, mapsIndoorsInstance, currentKioskLocation]);
+
+    /*
+     * When both origin location and destination location are selected, and have geometry, 
+     * call the MapsIndoors SDK to get information about the route.
+     */
+    useEffect(() => {
```

Comment:
I think we should put some effort into not having duplicate code. This is more or less the same as we have in the Wayfinding component. Can we extract it into a reusable component? Or even a custom hook?